pasqal_cloud/errors.py

```diff
@@ -1,4 +1,18 @@
-class BatchException(BaseException):
+from requests import HTTPError, Response
+
+
+class ExceptionWithResponseContext(BaseException):
+    def __init__(self, msg: str, e: HTTPError = None) -> None:
+        if not e:
+            return super().__init__(msg)
+        data = "without context."
+        resp: Response = e.response
+        if resp.content:
+            data = resp.json()
+        super().__init__(f"{msg}: {data}")
```

Comment:
Before the error rework, we had custom logic to parse the response body (something like look at `data` attribute if any, otherwise print the `message`). How does this compare to it?

Reply:
I''ll refactor to mirror what was done